beacon_node/http_api/tests/broadcast_validation_tests.rs

```diff
@@ -0,0 +1,1337 @@
+            *b.parent_root_mut() = Hash256::zero();
+        })
+        .await;
+
+    let blinded_block: SignedBlindedBeaconBlock<E> = block.into();
+
+    let response: Result<(), eth2::Error> = tester
+        .client
+        .post_beacon_blinded_blocks_v2(&blinded_block, validation_level)
+        .await;
+    assert!(response.is_err());
+
+    let error_response: eth2::Error = response.err().unwrap();
+
+    /* mandated by Beacon API spec */
+    assert_eq!(error_response.status(), Some(StatusCode::BAD_REQUEST));
+
+    assert!(
+        matches!(error_response, eth2::Error::ServerMessage(err) if err.message == "BAD_REQUEST: NotFinalizedDescendant { block_parent_root: 0x0000000000000000000000000000000000000000000000000000000000000000 }".to_string())
+    );
+}
+
+/// This test checks that a block that is only valid from a gossip perspective is rejected when using `broadcast_validation=consensus`.
+#[tokio::test(flavor = "multi_thread", worker_threads = 2)]
+pub async fn blinded_consensus_gossip() {
+    /* this test targets gossip-level validation */
+    let validation_level: Option<BroadcastValidation> = Some(BroadcastValidation::Consensus);
+
+    // Validator count needs to be at least 32 or proposer boost gets set to 0 when computing
+    // `validator_count // 32`.
+    let validator_count = 64;
+    let num_initial: u64 = 31;
+    let tester = InteractiveTester::<E>::new(None, validator_count).await;
+
+    // Create some chain depth.
+    tester.harness.advance_slot();
+    tester
+        .harness
+        .extend_chain(
+            num_initial as usize,
+            BlockStrategy::OnCanonicalHead,
+            AttestationStrategy::AllValidators,
+        )
+        .await;
+    tester.harness.advance_slot();
+
+    let slot_a = Slot::new(num_initial);
+    let slot_b = slot_a + 1;
+
+    let state_a = tester.harness.get_current_state();
+    let (block, _): (SignedBeaconBlock<E>, _) = tester
+        .harness
+        .make_block_with_modifier(state_a, slot_b, |b| *b.state_root_mut() = Hash256::zero())
+        .await;
+
+    let blinded_block: SignedBlindedBeaconBlock<E> = block.into();
+
+    let response: Result<(), eth2::Error> = tester
+        .client
+        .post_beacon_blinded_blocks_v2(&blinded_block, validation_level)
+        .await;
+    assert!(response.is_err());
+
+    let error_response: eth2::Error = response.err().unwrap();
+
+    /* mandated by Beacon API spec */
+    assert_eq!(error_response.status(), Some(StatusCode::BAD_REQUEST));
+
+    assert!(
+        matches!(error_response, eth2::Error::ServerMessage(err) if err.message == "BAD_REQUEST: Invalid block".to_string())
+    );
+}
+
+/// This test checks that a block that is valid from both a gossip and consensus perspective is accepted when using `broadcast_validation=consensus`.
+#[tokio::test(flavor = "multi_thread", worker_threads = 2)]
+pub async fn blinded_consensus_partial_pass_only_consensus() {
```

Comment:
this should be refactored to look like `consensus_partial_pass_only_consensus` and expect a 200